packages/openjdk-17/build.sh

```diff
@@ -86,10 +86,13 @@ termux_step_create_debscripts() {
 	binaries="$(find $TERMUX_PREFIX/lib/jvm/java-17-openjdk/bin -executable -type f | xargs -I{} basename "{}" | xargs echo)"
 	manpages="$(find $TERMUX_PREFIX/lib/jvm/java-17-openjdk/man/man1 -name "*.1.gz" | xargs -I{} basename "{}" | xargs echo)"
 
-	cp $TERMUX_PKG_BUILDER_DIR/hooks/$TERMUX_PACKAGE_FORMAT/{postinst,prerm} .
-	sed -i -e "s|@TERMUX_PREFIX@|${TERMUX_PREFIX}|g" \
-		-e "s|@binaries@|${binaries}|g" \
-		-e "s|@manpages@|${manpages}|g" ./{postinst,prerm}
+	for hook in postinst prerm; do
+		sed -e "s|@TERMUX_PREFIX@|${TERMUX_PREFIX}|g" \
+			-e "s|@binaries@|${binaries}|g" \
+			-e "s|@manpages@|${manpages}|g" \
+			"$TERMUX_PKG_BUILDER_DIR/hooks/$TERMUX_PACKAGE_FORMAT/$hook.in" > $hook
+		chmod 755 $hook
```

Comment:
This should ideally be `700`. That should help with reducing potential attack surface, like for users who disable selinux who some reason. Only `owner` user needs access for executing in termux. Termux directories including `bin` files are also the same. Not critical to be done now though.

`dpkg` should be patched as well at https://github.com/guillemj/dpkg/blob/b2f9600ead232a2dd3c27f8b52807a9ca5854d17/src/main/script.c#L88

Related https://github.com/termux/termux-packages/blob/48714d8a7a6637bf335c4c0ac7ca5275ea214c44/scripts/bootstrap/termux-bootstrap-second-stage.sh#L190

Reply:
Thanks, yeah, plenty of permissions to fix, let's address in another PR. Will update this PR while we are at it though